src/compile.ts

```diff
@@ -407,6 +472,291 @@ export function compile(
+              node.expression,
+              nativeExprContext.closureNode
+            );
+
+            if (!outOfScopeIntegrationReference) {
+              throw Error(
+                "Integration Variable must be defined out of scope: " +
+                  node.expression.getText()
+              );
+            }
+
+            // add the function identifier to the integrations
+            nativeExprContext.registerIntegration(
+              outOfScopeIntegrationReference
+            );
+            // call the integration call function with the prewarm context and arguments
+            // At this point, we know native will not be undefined
+            // await integration.native.call(args, preWarmContext)
+            // FIXME: doesn't work without an array
+            // TODO: Support both sync and async function invocations: https://github.com/sam-goodwin/functionless/issues/105
+            return [
+              context.factory.createAwaitExpression(
+                context.factory.createCallExpression(
+                  context.factory.createPropertyAccessExpression(
+                    context.factory.createPropertyAccessExpression(
+                      node.expression,
+                      "native"
+                    ),
+                    "call"
+                  ),
+                  undefined,
+                  [
+                    context.factory.createArrayLiteralExpression(
+                      node.arguments
+                    ),
+                    // TODO: determine if this is needed at all?
+                    nativeExprContext.preWarmContext,
+                  ]
+                )
+              ),
+            ];
+          }
+        } else if (ts.isNewExpression(node)) {
+          const newType = checker.getTypeAtLocation(node);
+          // cannot create new resources in native runtime code.
+          const functionlessKind = getFunctionlessTypeKind(newType);
+          if (getFunctionlessTypeKind(newType)) {
+            throw Error(
+              `Cannot initialize new resources in a native function, found ${functionlessKind}.`
+            );
+          }
+          // TODO: check for CDK constructs
+          // const type = checker.getTypeAtLocation(node.expression);
+          // if (isCDKConstruct(type)) {
+          //   throw Error(
+          //     `Cannot initialize new resources in a native function, found ${type.symbol.name}.`
+          //   );
+          // }
+        } else if (ts.isReturnStatement(node)) {
+          const child = node.expression
+            ? toNativeExpr(node.expression, context, nativeExprContext)
+            : undefined;
+          if (!child) {
+            return node;
+          } else if (Array.isArray(child)) {
+            // if multiple nodes are returned, assume the last one is the value to assign and the rest should be injected as siblings.
+            const [last, ...rest] = child.reverse();
```

Comment:
I don't understand?

Reply:
Need comments with examples.

Reply:
The compiler supports returning an array of nodes, but only when the parent supports multiple nodes. I needed to inject siblings into the context. If the child returns multiple nodes and the parent cannot handle multiple nodes (ex: `VariableAssignment`), they get passed up the tree until the parent can support multiple nodes (ex: `ExpressionStatement`)

For example, if I wanted to create a client that a line needs, I wanted to return the client along with the transformed node.

```ts
bus()
```

Compiled to

```ts
const EB = new EventBridge()
await EB.putTargets()
```

Howerever... the one use case I had for this is now supported using the prewarm context, so I don't need this anymore!